consensus/src/pipeline/virtual_processor/processor.rs

```diff
@@ -1188,6 +1191,190 @@ impl VirtualStateProcessor {
         Ok(())
     }
 
+    pub fn get_utxo_return_address(
```

Comment:
Maybe we should move this set of new methods to a new parallel file under `pipeline/virtual_processor` which will `impl VirtualStateProcessor` too.  We can call it `utxo_query.rs` or `utxo_query_manager.rs` and @D-Stacks will be able to extend it for his needs

Reply:
Maybe `utxo_inquirer.rs`